dev/tools/localization/localizations_validator.dart

```diff
@@ -88,6 +88,32 @@ void validateEnglishLocalizations(File file) {
     throw ValidationError(errorMessages.toString());
 }
 
+void removeUndefinedLocalizations(
```

Comment:
Add a doc comment. It looks like this:
1. Looks up the canonical (English, in this case) localizations.
2. For each locale, gets the resources.
3. Determines the set of keys that aren't plural variations (I presume because we're only interested in the base terms being translated and not their variants?)
4. Determines the set of invalid keys; that is those that are (non-plural) keys in the resources for this locale, but which _aren't_ keys in the canonical list.
5. Removes the invalid keys from this resource's locale.

Reply:
Thanks!